pinc/misc.inc

```diff
@@ -419,6 +419,65 @@ function get_numeric_param(string $type, array $arr, string $key, $default, $min
     }
 }
 
+/**
+ * Get and validate a boolean value from an array
+ *
+ * Accepts various reasonably "truthy" values, like 0, 1, "0", "1", "true",
+ * "false" and various capitalizations thereof.
+ *
+ * @param array<string, mixed> $arr
+ * @param string $key
+ * @param bool|null $default
+ * @param bool $allownull
+ *
+ * @return bool|null
+ *
+ * @throws InvalidArgumentException
```

Comment:
It would probably be better to either set the PHP types or PHPStan annotations, but not both. Having both could lead to inconsistencies between what PHPStan analyzes and the runtime behavior.

Here as we set the PHP types so we could simplify this to only the types that can't be represented by PHPStan:

```suggestion
 * @param array<string, mixed> $arr
 *
 * @throws InvalidArgumentException
```

Thoughts?

Reply:
We originally added these for (not yet completed, obviously) documentation not for phpstan and your suggestion will work here because these params don't have descriptions, but elsewhere we add param descriptions not just types. Any thoughts on how to handle this more broadly?

Reply:
I would say that the need for good documentation should trump the potential inconsistencies in PHPStan. Those are mostly theoretical at this point and we can revisit this if we find that they cause issues.

Here just adding the names+types of the parameters seems like it could be generated from the function signature (though I don't know if that's how PHPDoc work). Those with descriptions do add valuable documentation so we should keep those.